bycycle/utils.py

```diff
@@ -0,0 +1,90 @@
+"""Plotting utility functions."""
+
+import numpy as np
+
+###################################################################################################
+###################################################################################################
+
+
+def limit_df(df, fs, xlim):
+    """Restrict dataframe to be within time limits.
+
+    Parameters
+    ----------
+    df : pandas DataFrame
+        Dataframe output of :func:`~.compute_features`.
+    fs : float
+        Sampling rate, in Hz.
+    xlim : tuple of (float, float), optional, default: None
+        Start and stop times.
+
+    Returns
+    -------
+    df : pandas DataFrame
+        A limited dataframe of cycle features.
+    """
+
+    center_e, side_e = get_extrema(df)
+
+    # Limit dataframe to xlim
+    df = df[(df['sample_next_' + side_e].values >= xlim[0]*fs) &
+            (df['sample_last_' + side_e].values < xlim[1]*fs)]
+
+    # Shift sample indices to start at 0
+    df['sample_last_' + side_e] = df['sample_last_' + side_e] - int(fs * xlim[0])
+    df['sample_next_' + side_e] = df['sample_next_' + side_e] - int(fs * xlim[0])
+    df['sample_' + center_e] = df['sample_' + center_e] - int(fs * xlim[0])
+    df['sample_zerox_rise'] = df['sample_zerox_rise'] - int(fs * xlim[0])
+    df['sample_zerox_decay'] = df['sample_zerox_decay'] - int(fs * xlim[0])
+
+    return df
+
+
+def limit_sig_times(sig, times, xlim):
+    """Restrict signal and times to be within time limits.
+
+    Parameters
+    ----------
+    sig : 1d array
```

Comment:
what do we think about ordering for `sig` and `times`. In plot funcs, I think we generally have `times` first, then `sig`. It might make sense for this to follow that, and so reverse order?

Reply:
Whoops, yeah, that is the order in neurodsp. I'll change this.